lib/features/history/history_logic/repositories/allowance_repository.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:givt_app_kids/core/network/api_service.dart';
+import 'package:givt_app_kids/features/history/models/allowance.dart';
+
+mixin AllowanceHistoryRepository {
+  Future<List<Allowance>> fetchAllowanceHistory({required String childId});
+}
+
+class AllowanceHistoryRepositoryImpl with AllowanceHistoryRepository {
+  AllowanceHistoryRepositoryImpl(
+    this._apiService,
+  );
+
+  final APIService _apiService;
+
+  @override
+  Future<List<Allowance>> fetchAllowanceHistory(
+      {required String childId}) async {
+    final response =
+        await _apiService.fetchHistory(childId, 'RecurringAllowance');
+
+    List<Allowance> result = [];
```

Comment:
in regard of single repo: the idea is to have something like:
History class with date and amount, and Donation extends History with all other fields. In this case we can use single repo